spyder/utils/programs.py

```diff
@@ -817,84 +830,67 @@ def is_module_installed(module_name, version=None, installed_version=None,
-            get_modver = inspect.getsource(get_module_version)
-            stable_ver = inspect.getsource(is_stable_version)
-            ismod_inst = inspect.getsource(is_module_installed)
+    if installed_version is not None:
+        '''
+        installed_version only passed in from dependencies.Dependency.check
+        which retrieved it from get_module_version, so we know the module is
+        already installed
+        '''
+        return True
 
-            f = tempfile.NamedTemporaryFile('wt', suffix='.py',
-                                            dir=get_temp_dir(), delete=False)
+    module_version = None
+    if interpreter is not None:
+        if is_python_interpreter(interpreter):
+            cmd = 'import %s; print(%s.__version__)' % ((module_name,) * 2)
             try:
-                script = f.name
-                f.write("# -*- coding: utf-8 -*-" + "\n\n")
-                f.write("from distutils.version import LooseVersion" + "\n")
-                f.write("import re" + "\n\n")
-                f.write(stable_ver + "\n")
-                f.write(checkver + "\n")
-                f.write(get_modver + "\n")
-                f.write(ismod_inst + "\n")
-                if version:
-                    f.write("print(is_module_installed('%s','%s'))"\
-                            % (module_name, version))
-                else:
-                    f.write("print(is_module_installed('%s'))" % module_name)
+                # use clean environment
+                proc = run_program(interpreter, ['-c', cmd], env={})
+                output, _err = proc.communicate()
+            except Exception:
+                return False
 
-                # We need to flush and sync changes to ensure that the content
-                # of the file is in disk before running the script
-                f.flush()
-                os.fsync(f)
-                f.close()
-                try:
-                    proc = run_program(interpreter, [script])
-                    output, _err = proc.communicate()
-                except subprocess.CalledProcessError:
-                    return True
-                return eval(output.decode())
-            finally:
-                if not f.closed:
-                    f.close()
-                os.remove(script)
+            if _err.decode() == '':
+                module_version = output.decode().strip()
+            else:
+                return False
         else:
             # Try to not take a wrong decision if interpreter check
             # fails
             return True
-    else:
-        if installed_version is None:
-            try:
-                actver = get_module_version(module_name)
-            except:
-                # Module is not installed
-                return False
-        else:
-            actver = installed_version
-        if actver is None and version is not None:
+
+    if module_version is None:

```

Comment:
How can `module_version` be other than `None` if nothing has modified it between your declaration above and this point?

This looks ugly to me, so could you rewrite it a bit?

Reply:
Okay, I'll look at this.

Reply:
Done.